src/OpenTelemetry/Trace/TracerProviderSdk.cs

```diff
@@ -68,12 +68,7 @@ static TracerProviderSdk()
                 // Callback when Activity is started.
                 ActivityStarted = (activity) =>
                 {
-                    if (!activity.IsAllDataRequested)
-                    {
-                        return;
-                    }
-
-                    if (SuppressInstrumentationScope.IncrementIfTriggered() == 0)
+                    if (activity.IsAllDataRequested && !Sdk.SuppressInstrumentation)

```

Comment:
I think this part shouldn't be changed since there are new style activities generated from `ActivitySource.StartActivity`.

Reply:
You're correct! I removed the modification to this file.

Now, `SuppressInstrumentationScope` exposes two public methods `Begin()` and `Enter()`. Should be remove `Begin()` and `Dispose()` in a different PR?

Reply:
No, I think `Begin` and `Enter` has different semantics and both are useful, check out their XML comments.